src/index.ts

```diff
@@ -145,10 +138,35 @@ export default class SwupA11yPlugin extends Plugin {
 		};
 	}
 
-	prepareAnnouncement(visit: Visit) {
-		// Allow customizing announcement before this hook
-		if (typeof visit.a11y.announce !== 'undefined') return;
+	announceContent(visit: Visit) {
+		this.swup.hooks.callSync('content:announce', visit, undefined, (visit) => {
+			// Allow customizing announcement before this hook
+			if (typeof visit.a11y.announce === 'undefined') {
+				visit.a11y.announce = this.getDocumentTitle();
+			}
+
+			// Announcement disabled for this visit?
+			if (!visit.a11y.announce) return;
 
+			// Why the 100ms delay? see research at https://github.com/swup/a11y-plugin/pull/50
+			this.announcer.announce(visit.a11y.announce, 100);
+		});
+	}
+
+	focusContent(visit: Visit) {
+		this.swup.hooks.callSync('content:focus', visit, undefined, (visit) => {
+			// Focus disabled for this visit?
+			if (!visit.a11y.focus) return;
+
+			// Found and focused [autofocus] element? Return early
+			if (this.focusAutofocusElement() === true) return;
+
+			// Otherwise, find and focus actual content container
+			this.focusContentElement(visit.a11y.focus);
+		});
+	}
+
+	getDocumentTitle(): string | undefined {
```

Comment:
I find `getDocumentTitle` a bit misleading. That term usually means the `<title>` element of the document for me. How about `getPageTitle` or `getPageAnnouncement`?